### PR TITLE
Add new parameter for SIGv4 auth.

### LINKED ISSUE #41: [FEATURE] Support SSH tunnel with AWS_SIGv4 auth
**Is your feature request related to a problem?**
An OpenSearch cluster located into AWS VPC can be reached with SSH tunnel or VPN only.
AWS_SIGv4 authentication requires extra header to set if SSH tunnel is used.

**What solution would you like?**
Add optional configuration parameter when AWS_SIGv4 is selected.

**What alternatives have you considered?**
Use BASIC auth instead.

**Do you have any additional context?**
https://stackoverflow.com/a/67683064

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
### Description
Add new optional parameter for use with AWS_SIGv4 authentication to connect to a cluster via SSH tunnel (for example - OpenSearch cluster is a VPC endpoint).
This parameter is required to fix request header to comply SIGv4 requirements. [Ref: link](https://stackoverflow.com/a/67683064).

New UI config parameter + new connection string parameter:

image

### TODOs
- [x] Update docs
- [x] Test on Windows
- [x] Test on Mac
- [x] Test compatibility with older version
- [x] Update logging
- [x] Fix UI to show second word and/or rename
- [ ] Fix UI to auto-activate new field when SIGv4 auth type is chosen
- [x] Trim string prefix and postfix
- [x] Add fix for #39 
 
### Issues Resolved
Fixes #41, #39
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/sqlodbc/dlg_specific.h [hunk 1]
```diff
@@ -63,7 +64,6 @@ extern "C" {
 #define DEFAULT_DRIVERNAME "opensearchodbc"
 #define DEFAULT_DESC ""
 #define DEFAULT_DSN ""
-#define DEFAULT_REGION ""
```
Comment (acarbonetto) [inline src/sqlodbc/dlg_specific.h:66]: not used anywhere?

File: src/sqlodbc/dlg_wingui.c [hunk 2]
```diff
@@ -63,14 +63,17 @@ void SetAuthenticationVisibility(HWND hdlg, const struct authmode *am) {
         EnableWindow(GetDlgItem(hdlg, IDC_USER), TRUE);
         EnableWindow(GetDlgItem(hdlg, IDC_PASSWORD), TRUE);
         EnableWindow(GetDlgItem(hdlg, IDC_REGION), FALSE);
+        EnableWindow(GetDlgItem(hdlg, IDC_TUNNEL_HOST), FALSE);
```
Comment (acarbonetto) [inline src/sqlodbc/dlg_wingui.c:66]: we don't need this for basic auth?

File: src/sqlodbc/opensearch_communication.cpp [hunk 3]
```diff
@@ -432,14 +432,35 @@ OpenSearchCommunication::IssueRequest(
             credential_provider = Aws::MakeShared<
                 Aws::Auth::ProfileConfigFileAWSCredentialsProvider >(
                 ALLOCATION_TAG.c_str(), ESODBC_PROFILE_NAME.c_str());
+
+        LogMsg(OPENSEARCH_ALL,
+               (Aws::String("=== KEY = ")
+                + credential_provider->GetAWSCredentials().GetAWSAccessKeyId()
+                + " SECRET = "
+                + credential_provider->GetAWSCredentials().GetAWSSecretKey()
```
Comment (acarbonetto) [inline src/sqlodbc/opensearch_communication.cpp:440]: Don't do this unless it's anonymized.  We do not want to put AWS credentials in the log.

File: src/sqlodbc/opensearch_odbc.rc [hunk 4]
```diff
@@ -181,7 +183,7 @@ BEGIN
         BLOCK "040904e4"
         BEGIN
             VALUE "Comments", "OpenSearch ANSI ODBC driver"
-            VALUE "CompanyName", "Amazon"
+            VALUE "CompanyName", "OpenSearch Project"
```
Comment (acarbonetto) [inline src/sqlodbc/opensearch_odbc.rc:186]: 👍

**Review by acarbonetto — APPROVED** [review]

File: src/sqlodbc/dlg_wingui.c (hunk 2 above)
Comment (Yury-Fridlyand) [reply src/sqlodbc/dlg_wingui.c:66]: no

File: src/sqlodbc/opensearch_communication.cpp (hunk 3 above)
Comment (Yury-Fridlyand) [reply src/sqlodbc/opensearch_communication.cpp:440]: Sure, this was for debugging needs

File: src/sqlodbc/dlg_specific.h (hunk 1 above)
Comment (Yury-Fridlyand) [reply src/sqlodbc/dlg_specific.h:66]: It is duplicated by line 126

**Comment by acarbonetto** [comment]
Make sure to update documentation for connections and mention how to connect basic/sigv4 through a tunnel (and what to set as the tunnel host): https://github.com/opensearch-project/sql-odbc/blob/main/docs/user/configuration_options.md

**Review by matthewryanwells — APPROVED** [review]

**Review by dai-chen — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

**Review by dylanmtaylor — APPROVED** [review]
I have tested these changes and they fix an issue for us.

**Pull request merged**